src/cli/search.rs

```diff
@@ -93,24 +94,167 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         (None, None) => vec![Channel::from_str("conda-forge", &channel_config).into_diagnostic()?],
     };
 
-    let limit = args.limit;
-    let package_name = args.package;
+    let package_name_filter = args.package;
     let platforms = [Platform::current()];
     let repo_data = fetch_sparse_repodata(&channels, &platforms).await?;
 
-    let p = PackageName::try_from(package_name.clone()).into_diagnostic()?;
+    // When package name filter contains * (wildcard), it will search and display a list of packages matching this filter
+    if package_name_filter.contains('*') {
+        let package_name_without_filter = package_name_filter.replace("*", "");
+        let package_name = PackageName::try_from(package_name_without_filter).into_diagnostic()?;
+
+        let limit = args.limit;
+
+        let _ = search_package_by_wildcard(package_name, &package_name_filter, repo_data, limit)
+            .await?;
+    }
+    // If package name filter doesn't contain * (wildcard), it will search and display specific package info (if any package is found)
+    else {
+        let package_name = PackageName::try_from(package_name_filter).into_diagnostic()?;
+        let _ = search_exact_package(package_name, repo_data).await?;
+    }
+
+    Ok(())
+}
+
+async fn search_exact_package(
+    package_name: PackageName,
+    repo_data: Vec<SparseRepoData>,
+) -> miette::Result<()> {
+    let package_name_search = package_name.clone();
+    let packages = await_in_progress(
+        "searching packages",
+        spawn_blocking(move || {
+            search_package_by_filter(&package_name_search, &repo_data, |pn, n| {
+                pn == n.as_normalized()
+            })
+        }),
+    )
+    .await
+    .into_diagnostic()??;
+
+    if packages.is_empty() {
+        let normalized_package_name = package_name.as_normalized();
+        return Err(miette::miette!("Could not find {normalized_package_name}"));
```

Comment:
I'm questioning what would be the best thing to do here. Either automatically search for the name as if it was a filter. Or we should hint the user to use a wildcard. Otherwise this functionality is a bit hidden.

Reply:
Should I return a message, like "Package x not found, please use a wildcard '*' in your search if you want a broader result."?

Reply:
Yes that would be better.